environment/src/main/java/com/sequenceiq/environment/network/v1/converter/EnvironmentBaseNetworkConverter.java

```diff
@@ -97,6 +98,15 @@ protected Set<DeploymentRestriction> getDeploymentRestrictionForPrivateSubnet(Su
                 : DeploymentRestriction.NON_ENDPOINT_ACCESS_GATEWAYS;
     }
 
+    protected boolean isAllSubnetPublic(CreatedCloudNetwork createdCloudNetwork) {
+        return createdCloudNetwork.getSubnets().stream().filter(e -> e.isPublicSubnet()).collect(Collectors.toSet()).size()
```

Comment:
Can we use `allMatch` method of the Stream API instead of comparing primitive types?

Reply:
We can rename this to isAllSubnetsPublic